src/search/params.cc

```diff
@@ -30,16 +30,67 @@ SearchQuery::clear()
   void
 from_json( const nlohmann::json & jfrom, SearchQuery & qry )
 {
-  pkgdb::from_json( jfrom, dynamic_cast<pkgdb::PkgDescriptorBase &>( qry ) );
-  try { jfrom.at( "partialMatch" ).get_to( qry.partialMatch ); }
-  catch( const nlohmann::json::out_of_range & ) {}
+  auto getOrFail = [&]( const std::string    & key
```

Comment:
(I don't feel like you need to move it, but I do think this change probably belongs in a separate PR)

Reply:
I added it here since I was replacing the `from_json` invocation on the base class and wanted to get nicer messages.

For similar cases I think we'll tackle those in the "better error handling" PR.